pipeline/frontend/yaml/constraint/constraint.go

```diff
@@ -120,6 +127,16 @@ func (when *When) UnmarshalYAML(value *yaml.Node) error {
 // Match returns true if all constraints match the given input. If a single
 // constraint fails a false value is returned.
 func (c *Constraint) Match(metadata frontend.Metadata) bool {
+	// if event filter is not set, set default
+	if c.Event.IsEmpty() {
+		c.Event.Include = []string{
+			frontend.EventPush,
+			frontend.EventPull,
+			frontend.EventTag,
+			frontend.EventDeploy,
+		}
+	}
```

Comment:
As I wrote in my comment this feels like unexpected magic to me. There is no sign that cron works differently.

Reply:
well if we want more events like pull_closed etc ... if not filtered by default we should have pipelines that would look like they get exec unitended

Reply:
#1140